src/ppx_deriving.cppo.ml

```diff
@@ -12,6 +12,21 @@ open Parsetree
 open Ast_helper
 open Ast_convenience
 
+#if OCAML_VERSION >= (4, 05, 0)
+module Typ = struct
+  include Typ
+
+  let poly ?(loc= !Ast_helper.default_loc) ?attrs vars ty =
+    let vars = List.map (fun txt -> { Asttypes.loc; txt }) vars in
+    Typ.poly ~loc ?attrs vars ty
```

Comment:
Long-term it would be better to patch easier OCaml versions to provide the more precise version, and then use it in the file. I believe that only `free_vars_in_core_type` would have to be adapted to returned located strings (not difficult). I would be happy to provide a patch for this now or later.